src/Widgets/calendar/GridDay.vala

```diff
@@ -56,20 +57,65 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
 
         label.name = "date";
 
-        add (label);
+        main_grid = new Gtk.Grid ();
+        main_grid.halign = Gtk.Align.CENTER;
+        main_grid.orientation = Gtk.Orientation.VERTICAL;
+        main_grid.attach (label, 0, 0);
+
+        add (main_grid);
+        halign = Gtk.Align.CENTER;
         show_all ();
 
         // Signals and handlers
         button_press_event.connect (on_button_press);
         key_press_event.connect (on_key_press);
         scroll_event.connect ((event) => {return Util.on_scroll_event (event);});
+
+        var model = Widgets.CalendarModel.get_default ();
+        model.events_added.connect (update_event_days);
+        model.events_updated.connect (update_event_days);
+        model.events_removed.connect (update_event_days);
     }
 
     public void update_date (GLib.DateTime date) {
         this.date = date;
         label.label = date.get_day_of_month ().to_string ();
     }
 
+    public async void update_event_days () {
+        GLib.Idle.add (() => {
+            var events = Widgets.CalendarModel.get_default ().get_events (date);
+
+            var event_dot_grid = new Gtk.Grid ();
+            event_dot_grid.halign = Gtk.Align.CENTER;
+
+            var event_dot = new Gtk.Image ();
+            event_dot.pixel_size = 6;
+
+            if (event_dot_grid != null) {
+                event_dot_grid.destroy ();
+            }
+            if (events.size != 0) {
+                if (events.size <= 4) {
+                    foreach (var e in events) {
+                        if (e != null) {
+                            event_dot.gicon = new ThemedIcon ("pager-checked-symbolic");
+                            Util.set_event_calendar_color (e.cal, event_dot);
+                            event_dot_grid.add (event_dot);
+                        }
+                    }
+                } else if (events.size > 4) {
+                    event_dot.gicon = new ThemedIcon ("events-bar-symbolic");
+                    event_dot_grid.add (event_dot);
+                }
+                event_dot_grid.show_all ();
+                main_grid.margin_top = 5;
```

Comment:
Instead of adding a margin here, I think setting a height request for the event dot grid would be more reliable

Reply:
Reliable, but results in this:
![Screenshot from 2019-07-12 13 27 20](https://user-images.githubusercontent.com/4886639/61143564-eaaa4880-a4a8-11e9-8d07-1549102f0924.png)

I'll make some adjustments with Gtk.Align, and make it a non-magic number of 6.